src/pip/_internal/commands/download.py

```diff
@@ -222,13 +232,39 @@ def run(self, options, args):
                 resolver.resolve(requirement_set)
 
                 downloaded = ' '.join([
-                    req.name for req in requirement_set.successfully_downloaded
+                    req.name
+                    for req in requirement_set.successfully_downloaded
                 ])
                 if downloaded:
                     logger.info('Successfully downloaded %s', downloaded)
 
+                if options.json:
+                    details = self._get_download_details(
+                        resolver, requirement_set, options.download_dir)
+                    print(json.dumps(details))
```

Comment:
I'm concerned about this - we never use raw "print" statements anywhere else in pip to generate output (see `pip list` for a concrete example, which uses `logger.info`). This PR should use the same mechanisms as the rest of pip does.